src/integrationTests/async.spec.ts

```diff
@@ -23,201 +23,233 @@ beforeEach(() => {
   xhrMock = new mocks.XhrMock(sdkWindow);
   _cryptoMock = new mocks.CryptoMock(sdkWindow);
   mocks.setCookieMock(sdkWindow.document);
+  removeUid2Cookie();
+  removeUid2LocalStorage();
 });
 
 afterEach(() => {
   mocks.resetFakeTime();
 });
 
+const removeUid2Cookie = mocks.removeUid2Cookie;
+const removeUid2LocalStorage = mocks.removeUid2LocalStorage;
+
 const makeIdentity = mocks.makeIdentityV2;
 
-describe("when getAdvertisingTokenAsync is called before init", () => {
-  describe("when initialising with a valid identity", () => {
-    const identity = makeIdentity();
-    test("it should resolve promise after invoking the callback", () => {
-      const p = uid2.getAdvertisingTokenAsync().then((token: any) => {
-        expect(callback).toHaveBeenCalled();
-        return token;
-      });
-      uid2.init({ callback: callback, identity: identity });
-      jest.runAllTimers();
-      return expect(p).resolves.toBe(identity.advertising_token);
+let useCookie: boolean | undefined = undefined;
+
+const testCookieAndLocalStorage = (test: () => void, only = false) => {
+  const describeFn = only ? describe.only : describe;
+  describeFn('Using default: ', () => {
+    beforeEach(() => {
+      useCookie = undefined;
     });
+    test();
   });
-
-  describe("when initialising with an invalid identity", () => {
-    test("it should reject promise after invoking the callback", () => {
-      const p = uid2.getAdvertisingTokenAsync().catch((e: any) => {
-        expect(callback).toHaveBeenCalled();
-        throw e;
-      });
-      uid2.init({ callback: callback });
-      return expect(p).rejects.toBeInstanceOf(Error);
+  describeFn('Using cookies ', () => {
+    beforeEach(() => {
+      useCookie = true;
     });
+    test();
   });
-
-  describe("when initalising with a non-expired identity which requires a refresh", () => {
-    test("it should resolve updated advertising", () => {
-      const originalIdentity = makeIdentity({
-        refresh_from: Date.now() - 100000,
-      });
-      const updatedIdentity = makeIdentity({
-        advertising_token: "updated_advertising_token",
-      });
-      const p = uid2.getAdvertisingTokenAsync();
-      uid2.init({ identity: originalIdentity });
-      xhrMock.responseText = btoa(
-        JSON.stringify({ status: "success", body: updatedIdentity })
-      );
-      xhrMock.onreadystatechange(new Event(""));
-      return expect(p).resolves.toBe(updatedIdentity.advertising_token);
+  describeFn('Using local storage ', () => {
+    beforeEach(() => {
+      useCookie = false;
     });
+    test();
   });
+};
+
+testCookieAndLocalStorage(() => {
+  describe("when getAdvertisingTokenAsync is called before init", () => {
+    describe("when initialising with a valid identity", () => {
```

Comment:
Do we cover test cases for when initialising without identity?  (which calls the loadIdentityWithFallback?)

Reply:
Do we have any test that testing if the refreshed token has been stored to the correct storage?